x/oracle/types/exchange_rates.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"encoding/json"
+	time "time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewDenomExchangeRate creates a DenomExchangeRate instance
+func NewDenomExchangeRate(denom string, exchangeRate sdk.Dec, t time.Time) DenomExchangeRate {
+	return DenomExchangeRate{
+		Denom:     denom,
+		Rate:      exchangeRate,
+		Timestamp: t,
+	}
+}
+
+func (v DenomExchangeRate) String() string {
+	bz, _ := json.Marshal(v)
+	return string(bz)
+}
+
+// ExchangeRate is type for storing rate and timestamp of denom into store without denom.
+type ExchangeRate struct {
+	Rate      sdk.Dec   `json:"rate"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+// Marshal implements store.Marshalable.
+func (d *ExchangeRate) Marshal() ([]byte, error) {
+	return json.Marshal(d)
+}
```

Comment:
Do we need to worry about any edge cases here, such as `<nil>` sdk.Dec?

Perhaps add test